src/otx/algorithms/common/adapters/mmcv/pipelines/load_image_from_otx_dataset.py

```diff
@@ -0,0 +1,199 @@
+"""Pipeline element that loads an image from a OTX Dataset on the fly."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from tempfile import TemporaryDirectory
+from typing import Any, Dict, Optional, Tuple
+
+import numpy as np
+
+from otx.algorithms.common.utils.data import get_image
+from otx.core.data.caching import MemCacheHandlerError, MemCacheHandlerSingleton
+
+_CACHE_DIR = TemporaryDirectory(prefix="img-cache-")  # pylint: disable=consider-using-with
+
+# TODO: refactoring to common modules
+
+
+class LoadImageFromOTXDataset:
+    """Pipeline element that loads an image from a OTX Dataset on the fly.
+
+    Can do conversion to float 32 if needed.
+    Expected entries in the 'results' dict that should be passed to this pipeline element are:
+        results['dataset_item']: dataset_item from which to load the image
+        results['dataset_id']: id of the dataset to which the item belongs
+        results['index']: index of the item in the dataset
+
+    Args:
+        to_float32 (bool, optional): True to convert images to fp32. defaults to False.
+        enable_memcache (bool, optional): True to enable in-memory cache. defaults to True.
+    """
+
+    def __init__(self, to_float32: bool = False, enable_memcache: bool = True):
+        self._to_float32 = to_float32
+        self._enable_memcache = enable_memcache
+        try:
```

Comment:
Do we always construct the `self._mem_cache_handler` even if user set the `enable_memcache` as False?

Reply:
Mem-cache handler singleton is created in the task initialization so that I didn't changed the creation behavior, but it would be also nice to disable `get()` itself.